doc/concepts/promotion.rst

```diff
@@ -76,3 +98,14 @@ repository. You can use the following workflow to update your test:
 You can also use ``dune runtest --auto-promote``, which will
 automatically do the promotion.
 
+Automatically Promoting Rule Output Into the Source Tree
+--------------------------------------------------------
+
+Dune rules support a ``(mode promote)`` directive that will automatically
+copy their output into your source tree. This is more suitable for code or
+documentation generation flows where you want to check in the output to make
+it easier to browse or to remove dependencies on the code generation step for 
+packaging in opam.
```

Comment:
This last sentence is a little awkward. Does this say the same thing?

This approach suits code or documentation generation workflows where output needs to be checked to enable easier browsing or to eliminate dependencies on the code generation step during opam packaging.

If not, let’s work on it to clarify.